lib/cocoapods/target/pod_target.rb

```diff
@@ -688,5 +716,43 @@ def uses_modular_headers?(only_if_defines_modules = true)
     def create_build_settings
       BuildSettings::PodTargetSettings.new(self)
     end
+
+    # Computes the destination sub-directory in the sandbox
+    #
+    # @param  [Pathname] headers_sandbox
+    #         The sandbox where the header links should be stored for this
+    #         Pod.
+    #
+    # @param  [Sandbox::FileAccessor] file_accessor
+    #         The consumer file accessor for which the headers need to be
+    #         linked.
+    #
+    # @param  [Array<Pathname>] headers
+    #         The absolute paths of the headers which need to be mapped.
+    #
+    # @return [Hash{Pathname => Array<Pathname>}] A hash containing the
+    #         headers folders as the keys and the absolute paths of the
+    #         header files as the values.
+    #
+    def header_mappings(file_accessor, headers)
+      consumer = file_accessor.spec_consumer
+      header_mappings_dir = consumer.header_mappings_dir
+      dir = headers_sandbox
+      dir += consumer.header_dir if consumer.header_dir
+
+      mappings = {}
+      headers.each do |header|
+        next if header.to_s.include?('.framework/')
+
+        sub_dir = dir
+        if header_mappings_dir
+          relative_path = header.relative_path_from(file_accessor.path_list.root + header_mappings_dir)
+          sub_dir += relative_path.dirname
+        end
+        mappings[sub_dir] ||= []
+        mappings[sub_dir] << header
+      end
+      mappings
+    end
```

Comment:
not that this is an issue, but note that this will now be part of the public API of `PodTarget` - does it need to be? it feels like something the sandbox should handle

Reply:
That's a good point. I'm open to hearing what others think about moving it into the sandbox. I don't really have a strong preference about where it lives, I just moved it here since everywhere we needed the header mappings we were already iterating over the pod targets, so it made sense in my head at the time to just put it in here.

Reply:
This might explode a bit but passing `file_accessor` does not make sense as the `pod_target` is already a holder and owner of `file_accessors`.

Reply:
So `header_mappings` won't actually be a part of the public API of `PodTarget` since it's under `private`. It takes in a `file_accessor` since it's more of a helper used by the public functions `public_header_mappings_by_file_accessor` and `header_mappings_by_file_accessor`